config.py

```diff
@@ -31,8 +34,7 @@ def main():
             print('Cloning spack instance to: ' + args.idir)
             if args.version is None:
                 args.version = spack_version
-            clone_cmd = 'git clone git@github.com:spack/spack.git -b' + args.version + ' ' + args.idir + '/spack'   
-            os.system(clone_cmd)
+            os.system('git clone ' + spack_repo + ' -b {0} {1}'.format(args.version, os.path.join(args.idir, 'spack')))
```

Comment:
```suggestion
            os.system('git clone {repo} -b {branch} {target_dir}'.format(repo=spack_repo, branch=args.version, target_dir=os.path.join(args.idir, 'spack')))
```
This is probably the most readable way to format strings pre python 3.6.
We don't need to change it everywhere now, but I think it probably makes sense to gradually move towards this variant.